NOISSUE: Add sos debugging tips to docs (#1562)
openshift/microshift: docs/debugging_tips.md

> > See [Generating sos reports for technical support](https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/8/html-single/generating_sos_reports_for_technical_support/index) for more information on the `sos` utility usage.

Reviewer: We support RHEL 9 now

```suggestion
> See [Generating sos reports for technical support](https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/9/html/getting_the_most_from_your_support_experience/generating-an-sos-report-for-technical-support_getting-the-most-from-your-support-experience) for more information on the `sos` utility usage.
```

Author: Done.

> See [documentation](https://github.com/sosreport/sos) for further information on `sos` utility.

Sos tool is composed of different plugins for tailored retrieval of information from different applications.

> A MicroShift specific plugin has been added from sos version 4.5.1, being able to gather:

Reviewer: ```suggestion
A MicroShift specific plugin has been added from sos version 4.5.1 and it can gather the following data:
```

Author: Done.

* MicroShift configuration and version
* YAML output for cluster-wide and system namespaced resources
* OVNK information

> Plugin is automatically enabled as soon as it detects any of the following conditions:

Reviewer: ```suggestion
The plugin is automatically enabled as soon as it detects any of the following conditions:
```

> Plugin is automatically enabled as soon as it detects any of the following conditions:
> * MicroShift RPMs being installed.
> * MicroShift systemd service being present, whether running or not.
> * MicroShift ovnk pods running.

Reviewer: ```suggestion
The plugin is automatically enabled when it detects any of the following conditions:
* MicroShift RPMs are installed
* MicroShift systemd service is present, whether running or not
* MicroShift ovnk pods are running
```

Author: Done.

> For more information on the plugin design and output format, please check the [enhancement proposal](https://github.com/openshift/enhancements/blob/master/enhancements/microshift/microshift-supportability-tools.md).

Reviewer: I'm not sure we should reference the ER from the docs. It can be confusing because the plugin will evolve and eventually not be like described in the ER.

Author: I included it because there is quite some relevant details there. Arent the ER's supposed to evolve with the implementation, if changes are significant enough?

Reviewer: It's just, these are user docs and ERs are very technical. A developer should be able to find the ER without the reference.

The plugin is also available through the usual options to enable/disable them individually or by profile.
```bash

> $ sos report --list-profiles
> 
> sosreport (version 4.3)
> 
> The following profiles are available:
> 
>  boot            boot, devices, dracut, grub2, services, systemd, udev
>  cluster         conntrack
>  container       cgroups, containers_common, crio, podman, selinux
>  debug           hts, kdump
>  desktop         soundcard, x11
>  hardware        ata, block, crypto, devices, hardware, kernel, logs,
>                  memory, multipath, networking, networkmanager, ntb,
>                  openhpi, pci, processor, scsi, soundcard, teamd, udev,
>                  udisks, usb, x11
>  identity        krb5, ldap, login, nis, pam, ssh, sssd, sudo
>  java            jars
>  kernel          ebpf, kernel, system
>  memory          memory, processor
>  microshift      microshift, microshift_ovn
>  network         conntrack, ebpf, firewall_tables, firewalld,
>                  networking, networkmanager, nss, openssl, teamd,
>                  unbound
>  openshift       selinux
>  packagemanager  dnf, rpm, subscription_manager, yum
>  performance     tuned
>  perl            perl
>  sap             vhostmd
>  security        auditd, nss, openssl, pam, selinux, ssh, sssd
>  services        chrony, nis, ssh, sssd, systemd, sysvipc, unbound
>  storage         ata, block, devicemapper, devices, filesys, iscsi,
>                  logs, lvm2, md, multipath, scsi, xfs
>  sysmgmt         dnf, insights, ldap, subscription_manager, yum
>  system          anaconda, anacron, auditd, boot, cgroups, chrony,
>                  console, cron, crypto, dbus, devices, dnf, ebpf,
>                  firewall_tables, hardware, host, i18n, insights, kdump,
>                  kernel, krb5, kvm, ldap, libraries, libvirt, login,
>                  logrotate, logs, memory, multipath, networking,
>                  networkmanager, openhpi, pam, pci, process, processor,
>                  python, release, rpm, selinux, services, ssh,
>                  subscription_manager, sudo, system, systemd, sysvipc
>                  tuned, udev, udisks, unbound, vhostmd, yum
>  virt            kvm, libvirt, vhostmd, xen
>  webserver       perl
> 
>  24 profiles, 91 plugins

Reviewer: Let's shorten the list. Alternatively, just grep for microshift to show the appropriate line

```suggestion
$ sos report --list-profiles

sosreport (version 4.3)

The following profiles are available:

 ...
 ...
 microshift      microshift, microshift_ovn
 ...
 ...

 24 profiles, 91 plugins
```

Author: Done.

```

Log into the host running MicroShift and execute the following command to generate an obfuscated
report that should not contain sensitive information (in this example only microshift plugin was

> enabled. Your system's output may vary):

Reviewer: Do you want to update the command below to explicitly run microshift plugin only? This way, we should not need to explain about different output, etc.

Author: Done.

> ```bash

Reviewer: Can we please shorten the output? Let's think what is relevant for the users to see. We may even want to skip it completely.

Author: Done.

        Commercial Support   : https://www.access.redhat.com/

The generated archive may contain data considered sensitive and its
content should be reviewed by the originating organization before being
passed to any third party.

No changes will be made to system configuration.

 Setting up archive ...
 Setting up plugins ...
 Running plugins. Please wait ...

  Starting 1/1   microshift      [Running: microshift]

  Finished running plugins

Found 1 total reports to obfuscate, processing up to 4 concurrently

sosreport-microshift-2-2023-03-23-ylwbkjc :        Beginning obfuscation...
sosreport-microshift-2-2023-03-23-ylwbkjc :        Obfuscation completed

Successfully obfuscated 1 report(s)

Creating compressed archive...

A mapping of obfuscated elements is available at
	/var/tmp/sosreport-microshift-2-2023-03-23-ylwbkjc-private_map

Your sosreport has been generated and saved in:
	/var/tmp/sosreport-microshift-2-2023-03-23-ylwbkjc-obfuscated.tar.xz

 Size	4.48KiB
 Owner	root
 sha256	282c06266297396aeeb1a21579811ac0f2db43d27bee9cf1cd55b997c952cb24

Please send this file to your support representative.
```
> Sos must always run with root privileges.

> The output file was generated in `/var/tmp/` dir, and a checksum file is also generated.

Reviewer: ```suggestion
The output file and its checksum are generated in `/var/tmp/` directory.
```

Author: Done.

```bash

> $ sudo ls -tr /var/tmp/

Reviewer: Do we want to leave the original command to only display the last two archives? The current command will display everything in the /var/tmp

Author: Fair enough, it should be mostly empty though.
